helsingborg-stad/app-mitt-helsingborg
# Refactor Input component

## Explain the changes you’ve made

Made some minor changes to the Input component, but the larger job was to change the places in SummaryList, EditableList and Repeater, so that we use a styled version of our input, rather than styling the React-Native input 'from scratch' every time. 

## Explain why these changes are made
While working on adding the done-button, I realised that we were repeating ourselves quite a bit, adding the same props and styling at each place. Adding the props for the done-button takes 3 lines, and adding those 3 lines in 10 separate places felt a bit too stupid and prone to bugs, so this refactoring felt very justified.  

## Explain your solution
One issue with using the Input inside the lists was that it displays the validation error message under it, which messes with the styling of the lists. So to fix this I add an optional prop, showErrorMessage, which can be passed when we want to see the error message. Similarly, I add an optional prop, transparent, which turns the background transparent. 

In the other files, I just change so that we style our input component, instead of the react native one. 
Inspired by uncle Bob, I also did a little comment cleaning. 

## How to test the changes?
Checkout out this branch, and go into stories or forms and try the text inputs. Everything should work and look as before.
